Final math commit

rnpark opened the pull request:
I think I resolved the merge conflicts, but best to double check

mholman5721 commented:
Hello, 

Go ahead and remove the file BeginExerciseActivity.java.orig

You can do this locally (just delete it), re-commit, and push up to github.

Meld generates these as a sort of backup. You can keep them locally, if you want, but generally they are undesirable for the master branch.

Secondly, the regex function you use in 'BeginExerciseActivity' is for Euler angles. The one for quaternions is in 'MainActivity.java' and is just called 'regEx'. If you have previous work using the quaternion version go ahead and include it as well (albeit commented out and marked with what you did).

Thanks.

rnpark commented:
I removed the file.  I though I had removed all the .orig files but I guess I missed one.

I don't understand what you're saying about the regex function.  Most of the math is being done on the feather and the quaternions are being converted back to Euler angles before being sent to the app.  I did it that way since I was going to need to convert them anyways and there was a function for that in the code for the bno055 sensors.  The math of the app side is mostly just some offsets to get the output in the ranges the arm is usually measured in.

mholman5721 commented:
There are actually two regex functions in the original function, 'regEx' and 'regexEuler' the only one taking in data to the app is 'regexEuler'. If you have work on a different repository with quaternion work, then that will need to be included as well in the final submission.

Thanks.
